dataset/models/tf/inception_v1.py

```diff
@@ -0,0 +1,229 @@
+
+    batch_norm : bool
+        if True enable batch normalization layers
+
+    data_format: str {'channels_last', 'channels_first'}
+
+    dropout_rate: float
+
+    head_type: str {'dense', 'conv'}
+         determine head of model
+
+         'dense' - consist of GAP and dense layers
+
+         'conv' - consist of conv 1x1 and GAP layers
+
+    """
+    def _build(self, *args, **kwargs):
+        _ = args
+
+        dim = self.get_from_config('dim', 2)
+        b_norm = self.get_from_config('batch_norm', True)
+        dropout_rate = self.get_from_config('dropout_rate', 0)
+        head_type = self.get_from_config('head_type', 'dense')
+        names = ['images', 'labels']
+        _, transformed_placeholders = self._make_inputs(names)
+
+        num_classes = self.num_classes('labels')
+        data_format = self.data_format('images')
+
+        max_pool = {'pool_size': 3,
+                    'strides': 2,
+                    'padding': 'same',
+                    'data_format': data_format}
+
+        batch_norm = {'training': self.is_training}
+        conv = {'data_format': data_format}
+
+        kwargs = {'max_pooling': max_pool,
+                  'batch_norm': batch_norm,
+                  'conv': conv,
+                  'data_format': data_format,
+                  'is_training': self.is_training}
+
+        with tf.variable_scope('inception'):
+            net = self.body(dim, transformed_placeholders['images'], b_norm, **kwargs)
+            net = self.head(dim, net, num_classes, head_type, data_format, self.is_training, dropout_rate)
+        
+        self.statistic(tf.identity(net, name='predictions'), transformed_placeholders['labels'])
+
+    @staticmethod
+    def body(dim, inputs, b_norm, **kwargs):
+        """ Building block for inception_v1 network
+
+        Parameters
+        ----------
+        dim: int
+
+        inputs: tf.Tensor
+
+        b_norm: bool:
+            if True enable batch normalization
+
+        Returns
+        -------
+        net: tf.Tensor
+        """
+        layout = 'cnpcncnp' if b_norm else 'cpccp'
+        with tf.variable_scope('body'):
+            print(layout)
+            net = conv_block(dim=dim, inputs=inputs, filters=[64, 64, 192], kernel_size=[7, 3, 3], strides=[2, 1, 1],\
+                             layout=layout, name='conv', **kwargs)
+            for i in range(len(FILTERS)):
+                length = len(FILTERS[i])
+                for name, filt in zip(NAMES[:length], FILTERS[i]):
+                    net = InceptionV1.block(dim=dim, inputs=net, filters=filt, name=str(i+3)+name,\
+                                            b_norm=b_norm, **kwargs)
+                    print(name, filt, net)
+                if i != 2:
+                    net = max_pooling(dim=dim, inputs=net, **kwargs['max_pooling'])
+        return net
+
+    @staticmethod
+    def head(dim, inputs, n_outputs, head_type='dense', data_format='channels_last', is_training=True,\
```

Comment:
You have `head` in the method name already, so don't repeat it in the parameter name